crates/core/src/addon.rs

```diff
@@ -371,7 +372,7 @@ impl Addon {
         } else if alpha_exists {
             3
         } else {
-            unreachable!("No file in curse package for {}", package.id);
+            return None;
```

Comment:
This is for the backlog.
@tarkah I have taken a quick decision here to return a `Optional<Addon>` because if people had a retail addon inside classic folder this would actually reach the unreachable code. And crash.

I think we can make it prettier than having the entire `fn` and `Optional` but in the light of speed, i choose this direction.